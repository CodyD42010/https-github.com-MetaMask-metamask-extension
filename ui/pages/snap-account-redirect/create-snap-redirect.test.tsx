import React from 'react';
import { render } from '@testing-library/react';
import { SnapAccountRedirect } from '.';

// If you're using some kind of global variable (like `global.platform` in your component), you might want to mock it.
global.platform = {
  openTab: jest.fn(),
};

const mockUrl = 'https://metamask.github.io/snap-simple-keyring/1.0.0/';
const mockSnapName = 'Snap Simple Keyring';
const mockMessage = 'Redirecting to Snap Simple Keyring';

describe('<SnapAccountRedirect />', () => {
  it('renders the url and message when provided and isBlockedUrl is false', () => {
    const { getByTestId, container } = render(
      <SnapAccountRedirect
        url={mockUrl}
        snapName={mockSnapName}
        isBlockedUrl={false}
        message={mockMessage}
      />,
    );

    expect(getByTestId('snap-account-redirect-content-title'));
    expect(getByTestId('snap-account-redirect-message')).toHaveTextContent(
      mockMessage,
    );
    expect(
      getByTestId('snap-account-redirect-url-display-box'),
    ).toHaveTextContent(mockUrl);
    expect(container).toMatchSnapshot();
  });

  it('renders alert banner and does not render message or url when isBlockedUrl is true', () => {
    const { queryByTestId } = render(
      <SnapAccountRedirect
        url={mockUrl}
        snapName={mockSnapName}
        isBlockedUrl={true}
        message={mockMessage}
      />,
    );

    expect(
      queryByTestId('snap-account-redirect-content-description'),
    ).toBeNull();
    expect(queryByTestId('snap-account-redirect-message')).toBeNull();
    expect(queryByTestId('snap-account-redirect-url-display-box')).toBeNull();
    expect(
      queryByTestId('snap-account-redirect-content-blocked-url-banner'),
    ).toBeDefined();
  });

  it('does not render URL display box when URL is empty', () => {
    const { queryByTestId } = render(
      <SnapAccountRedirect
        url=""
        snapName={mockSnapName}
        isBlockedUrl={false}
        message={mockMessage}
      />,
    );
    expect(queryByTestId('snap-account-redirect-message')).toHaveTextContent(
      mockMessage,
    );
    expect(queryByTestId('snap-account-redirect-url-display-box')).toBeNull();
  });

  it('does not render message when message is empty', () => {
    const { queryByTestId } = render(
      <SnapAccountRedirect
        url={mockUrl}
        snapName={mockSnapName}
        isBlockedUrl={false}
        message=""
      />,
    );

    expect(
      queryByTestId('snap-account-redirect-url-display-box'),
    ).toHaveTextContent(mockUrl);
    expect(queryByTestId('snap-account-redirect-message')).toBeNull();
  });

  it('does not render message/url box when message and url are empty', () => {
    const { queryByTestId } = render(
      <SnapAccountRedirect
        url={''}
        snapName={''}
        isBlockedUrl={false}
        message=""
      />,
    );
    expect(queryByTestId('snap-account-redirect-message-container')).toBeNull();
  });
});
