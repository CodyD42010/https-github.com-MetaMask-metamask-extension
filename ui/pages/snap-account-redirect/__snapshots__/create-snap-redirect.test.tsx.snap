// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SnapAccountRedirect /> renders the url and message when provided and isBlockedUrl is false 1`] = `
<div>
  <div
    class="mm-box create-snap-account-page mm-box--padding-right-4 mm-box--padding-left-4 mm-box--flex-direction-column mm-box--align-items-center mm-box--width-full mm-box--height-full mm-box--border-style-none"
  >
    <div
      class="mm-box mm-box--display-flex mm-box--flex-direction-column mm-box--align-items-center mm-box--height-full"
    >
      <div
        class="mm-box mm-box--padding-top-4 mm-box--display-flex mm-box--flex-direction-row mm-box--justify-content-space-between"
      >
        <div
          class="mm-box mm-box--display-flex mm-box--gap-4 mm-box--flex-direction-column mm-box--align-items-center"
        >
          <h2
            class="mm-box mm-text mm-text--heading-lg mm-text--text-align-center mm-box--color-text-default"
            data-testid="snap-account-redirect-content-title"
          >
            [snapAccountRedirectFinishSigningTitle]
          </h2>
          <p
            class="mm-box mm-text mm-text--body-md mm-text--text-align-center mm-box--color-text-default"
            data-testid="snap-account-redirect-content-description"
          >
            [snapAccountRedirectSiteDescription]
          </p>
          <div
            class="mm-box snap-delineator__wrapper mm-box--display-flex mm-box--flex-direction-column mm-box--background-color-background-default mm-box--rounded-lg mm-box--border-style-solid mm-box--border-color-border-default box--border-width-1"
          >
            <div
              class="mm-box snap-delineator__header mm-box--padding-1 mm-box--display-flex mm-box--justify-content-space-between mm-box--align-items-center"
              style="border-bottom-width: 1px;"
            >
              <div
                class="mm-box mm-box--display-flex mm-box--align-items-center"
              >
                <div
                  class="mm-box mm-text mm-avatar-base mm-avatar-base--size-xs mm-avatar-icon mm-text--body-xs mm-text--text-transform-uppercase mm-box--display-flex mm-box--justify-content-center mm-box--align-items-center mm-box--color-primary-default mm-box--background-color-info-default mm-box--rounded-full mm-box--border-color-transparent box--border-style-solid box--border-width-1"
                >
                  <span
                    class="mm-box mm-icon mm-icon--size-xs mm-box--display-inline-block mm-box--color-info-inverse"
                    style="mask-image: url('./images/icons/snaps.svg');"
                  />
                </div>
                <p
                  class="mm-box mm-text snap-delineator__header__text mm-text--body-sm mm-box--margin-top-0 mm-box--margin-bottom-0 mm-box--margin-left-1 mm-box--color-text-alternative"
                >
                  [contentFromSnap]
                </p>
              </div>
            </div>
            <div
              class="mm-box snap-delineator__content mm-box--padding-4 mm-box--display-flex mm-box--flex-direction-column"
            >
              <p
                class="mm-box mm-text mm-text--body-md mm-box--color-text-default"
                data-testid="snap-account-redirect-message"
              >
                Redirecting to Snap Simple Keyring
              </p>
              <div
                class="mm-box mm-box--padding-top-2 mm-box--display-flex"
              >
                <div
                  class="mm-box mm-box--padding-right-4 mm-box--display-inline-flex mm-box--align-items-center mm-box--background-color-background-default mm-box--rounded-sm mm-box--border-color-border-default mm-box--border-width-1 box--border-style-solid"
                >
                  <p
                    class="mm-box mm-text mm-text--body-md mm-box--padding-2 mm-box--color-primary-default"
                    data-testid="snap-account-redirect-url-display-box"
                  >
                    https://metamask.github.io/snap-simple-keyring/1.0.0/
                  </p>
                  <button
                    aria-label=""
                    class="mm-box mm-button-icon mm-button-icon--size-sm mm-box--display-inline-flex mm-box--justify-content-center mm-box--align-items-center mm-box--color-primary-default mm-box--background-color-transparent mm-box--rounded-lg"
                  >
                    <span
                      class="mm-box mm-icon mm-icon--size-sm mm-box--display-inline-block mm-box--color-inherit"
                      style="mask-image: url('./images/icons/export.svg');"
                    />
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
